hplsql/src/main/java/org/apache/hive/hplsql/functions/FunctionString.java

```diff
@@ -33,38 +33,30 @@ public FunctionString(Exec e, QueryExecutor queryExecutor) {
   public void register(BuiltinFunctions f) {
     f.map.put("CONCAT", this::concat);
     f.map.put("CHAR", this::char_);
-    f.map.put("INSTR", this::instr);
     f.map.put("LEN", this::len);
-    f.map.put("LENGTH", this::length);
-    f.map.put("LOWER", this::lower);
-    f.map.put("REPLACE", this::replace);
-    f.map.put("SUBSTR", this::substr);
-    f.map.put("SUBSTRING", this::substr);
     f.map.put("TO_CHAR", this::toChar);
     f.map.put("UPPER", this::upper);
-    
-    f.specMap.put("SUBSTRING", this::substring);
-    f.specMap.put("TRIM", this::trim);
```

Comment:
As I see from the test this is required. Please check the test. Put this back or change the test accordingly.

Reply:
@aturoczy Thanks for the review. Test cases are already added for SUBSTRING and TRIM, also for all functions.
https://github.com/apache/hive/pull/4965/commits/7c0a9f31477453d9974472f487dde4bfff577989#diff-4b45b06ec5de20b8929a80f493ead0990f1208e725d9a403774ecc5f27928227R798-R807
https://github.com/apache/hive/pull/4965/commits/7c0a9f31477453d9974472f487dde4bfff577989#diff-4b45b06ec5de20b8929a80f493ead0990f1208e725d9a403774ecc5f27928227R849-R852

Reply:
@mdayakar  I suppose the reason for removing SUBSTRING, TRIM and a few other functions is that these are builtin functions in Hive, so is HPL/SQL going to forward the function call to HS2 ? If so,  it would be good to add comments in some common class regarding this.   One thing we should verify is if the HPL/SQL behavior of these functions is the same as what the Hive built-in function does or is there any small difference.  I am curious why these were implemented in HPL/SQL in the first place since these are SQL standard functions.  Was there additional precondition checks being done or any other variants that HPL/SQL supported that Hive did not ?

Reply:
> I suppose the reason for removing SUBSTRING, TRIM and a few other functions is that these are builtin functions in Hive.
Yes Aman.

> so is HPL/SQL going to forward the function call to HS2 ?
Yes, in this case it will just forward the function call to HS2

> One thing we should verify is if the HPL/SQL behavior of these functions is the same as what the Hive built-in function does or is there any small difference.
Yes Aman, if there is any behaviour change or some extra feature for any function then that code is not removed. Actually for SUBSTRING also later found that it supports another functionality 'SUBSTRING(string FROM start_pos [FOR substring_len])' which is not present in HS2 so added back the SUBSTRING function code. Similar for CONCAT (if any param is NULL then HS2 returns NULL as output where as HPLSQL function ignores NULLs and concats other non-null strings) so this function also not removed.

>  I am curious why these were implemented in HPL/SQL in the first place since these are SQL standard functions.
As per my understanding HPLSQL initially provided as a separate entity (not part of Hive) and later it got merged as a part of Hive. When it was a separate entity, it can be used not only with Hive also can be with any other Databases like DB2, MySQL,... so they might provided implementation for all these functions now its part of Hive release so I assumed it can be used only with Hive so I removed the functions implementation which are already supported in Hive.

> Was there additional precondition checks being done or any other variants that HPL/SQL supported that Hive did not ?
The code is already present and I could see there are some precondition checks present.